blockchain/chain.go

```diff
@@ -511,6 +511,52 @@ func (b *BlockChain) getPrevNodeFromNode(node *blockNode) (*blockNode, error) {
 	return prevBlockNode, err
 }
 
+// ancestorNode returns the ancestor block 'distance' blocks before the passed
+// anchor block. While iterating backwards through the chain, any block nodes
+// which aren't in the memory chain are loaded in dynamically.
+//
+// This function MUST be called with the chain state lock held (for writes).
+func (b *BlockChain) ancestorNode(anchor *blockNode, distance uint32) (*blockNode, error) {
```

Comment:
I believe this will conflict with #755 which has the same function name, but different semantics.  This one goes back a certain distance while the one in #755 goes to a specific height.

We'll need to rename one or the other.  I'm open to suggestions for naming.